### PR TITLE
[WFLY-17514] NPE when -ds.xml file included in deployment contains va…

### PR DESCRIPTION
**Pull request by simkam**
…lidation configuration

https://issues.redhat.com/browse/WFLY-17514

### CODE REVIEW

**Review by tadamski — APPROVED** [review]

**Comment by jamezp** [comment]
this is ok to test

**Comment by simkam** [comment]
AFAICT tests failures are unrelated to this

File: connector/src/main/java/org/jboss/as/connector/subsystems/datasources/XMLDataSourceRuntimeHandler.java [hunk 1]
```diff
@@ -307,7 +307,10 @@ else if (attributeName.equals(Constants.EXCEPTION_SORTER_MODULE.getName())) {
             if (dataSource.getValidation().getExceptionSorter() == null) {
                 return;
             }
-            setStringIfNotNull(context, ((ModuleClassLoader)dataSource.getValidation().getExceptionSorter().getClassLoader()).getModule().toString());
+            ModuleClassLoader exceptionSorterClassLoader = (ModuleClassLoader) dataSource.getValidation().getExceptionSorter().getClassLoader();
+            if (exceptionSorterClassLoader != null) {
+                setStringIfNotNull(context, exceptionSorterClassLoader.getModule().toString());
+            }
```
Comment (bstansberry) [inline connector/src/main/java/org/jboss/as/connector/subsystems/datasources/XMLDataSourceRuntimeHandler.java:313]: For all of these, this code pattern is more robust, as it avoids the NPE but also doesn't assume the classloader type:

```
ClassLoader exceptionSorterClassLoader = dataSource.getValidation().getExceptionSorter().getClassLoader();
if (exceptionSorterClassLoader instanceof ModuleClassLoader) {
   setStringIfNotNull(context, ((ModuleClassLoader)exceptionSorterClassLoader).getModule().toString());
}
```

**Review by bstansberry — CHANGES_REQUESTED** [review]

File: connector/src/main/java/org/jboss/as/connector/subsystems/datasources/XMLDataSourceRuntimeHandler.java (hunk 1 above)
Comment (simkam) [reply connector/src/main/java/org/jboss/as/connector/subsystems/datasources/XMLDataSourceRuntimeHandler.java:313]: updated

**Review by bstansberry — APPROVED** [review]

**Pull request merged**